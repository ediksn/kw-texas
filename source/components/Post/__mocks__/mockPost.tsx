const mockPost = {
  id: 1,
  author: {
    name: 'Nina Royals',
    picture: '__mocks__/avatarNinaRoyals.png'
  },
  content:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean tellus leo, fringilla at elit sed, tincidunt viverra dui. Nullam dignissim mi et tincidunt luctus. Nullam ut eleifend diam. Phasellus faucibus aliquam leo, vitae scelerisque nunc dignissim in. Maecenas vulputate quam id metus efficitur tristique. Ut maximus varius enim eleifend feugiat. Ut at ultricies lorem, eget molestie risus. Nam ultricies tempus sem quis fermentum. Sed a eros ac metus scelerisque hendrerit. Ut nec urna aliquam, sagittis eros non, bibendum risus. Ut congue eleifend velit eu dictum. Praesent eu mi vitae leo pulvinar ullamcorper. Sed tortor felis, tristique vitae lacinia sit amet, pellentesque ut felis. Nulla finibus erat libero, eu sodales metus auctor at. Duis iaculis leo vitae odio mollis, feugiat aliquam erat eleifend. Mauris quis dui at lorem tincidunt viverra nec vel mauris. Pellentesque laoreet dapibus purus, eget vehicula nunc scelerisque quis. Donec ac nibh sed erat consequat sagittis ut ut odio. Curabitur tempor vitae neque vel gravida. Morbi varius ligula nec tellus accumsan, quis pharetra ante dictum. Nulla pharetra ac ex id cursus. Vivamus ut lacinia nunc. Maecenas nec orci risus.',
  date: '4 hours ago',
  likes: 13,
  comments: 5,
  shares: 2
}

export default mockPost
