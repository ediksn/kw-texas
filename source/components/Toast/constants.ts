import { verticalScale } from 'react-native-size-matters'

export const TOAST_HEIGHT = verticalScale(48)
export const TOAST_TYPE = {
  ERROR: 'ERROR',
  SUCCESS: 'SUCCESS'
}
