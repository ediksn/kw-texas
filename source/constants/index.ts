export { FORM } from './form'
export { NAVIGATION } from './navigation'
export { theme } from './theme'
export { ACTION } from './action'
export * from './statis'
export * from './testIds'
