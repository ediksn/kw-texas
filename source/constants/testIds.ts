export const TEST_IDS = {
  GENERAL_COMPONENTS: {
    VIDEO_CARD_ID: 'videoCardId',
    VIDEO_LIST_ID: 'videoListId',
    EVENT_CARD_ID: 'eventCardId',
    IMAGE_PICKER_PREVIEW_ID: 'imagePickerPreviewId',
    TOAST_ID: 'toastId'
  },
  EVENTS: {
    TODAY_ID: 'eventsTodayId',
    TOMORROW_ID: 'eventsTomorrowId'
  },
  EVENT_CARD: {
    IMAGE_ID: 'eventCardImageId',
    DATE_ID: 'eventCardImageId',
    TITLE_ID: 'eventCardTitleId',
    ASSISTANCE_ID: 'eventCardAssistanceId',
    INTERESTED_BUTTON_ID: 'eventCardInterestedButtonId',
    SHARE_BUTTON_ID: 'eventCardShareButtonId'
  },
  IMAGE_PICKER_PREVIEW: {
    IMAGE_ID: 'imageId'
  },
  GROUPS: {
    GROUP_ID: 'groupId',
    GROUP_SLIDER_ID: 'groupSliderId'
  },
  TOAST: {
    ICON_ID: 'toastIconId',
    TITLE_ID: 'toastTitleId'
  }
}

export const usernameInput = 'username-input'
export const passwordInput = 'password-input'
export const signinButton = 'signin-button'
export const forgotButton = 'forgot-button'
export const pickPrompts = 'pickPrompts-component'
export const kwLogo = 'kw-logo'
export const LoginKWLogo = 'login-kw-logo'
export const LoginBottomLogo = 'login-bottom-logo'
export const illustrationLogo = 'illustration-logo'
export const avatarPost = 'avatar-post'
export const authorPost = 'author-post'
export const datePost = 'date-post'
export const dotsOptionsPost = 'dotsoptions-post'
export const contentPost = 'content-post'
export const buttonPost = 'button-post'
export const linksPost = 'links'
