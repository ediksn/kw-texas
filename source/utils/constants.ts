export const MAX_CHARACTERS_NEW_POST: number = 500
export const POST_PREVIEW_MEDIA_LENGTH: number = 3
export const GROUP_PREVIEW_MEDIA_LENGTH: number = 10
