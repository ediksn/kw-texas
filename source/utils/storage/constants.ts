export default {
  LOGIN: {
    USER: 'LOGIN.USER',
    SESSION: 'LOGIN.SESSION'
  },
  BIOMETRIC: {
    PERMISSION: 'BIOMETRIC.PERMISSION',
    CREDENTIALS: 'BIOMETRIC.CREDENTIALS'
  }
}
