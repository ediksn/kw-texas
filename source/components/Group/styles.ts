import { StyleSheet } from 'react-native'
import { moderateScale, scale } from 'react-native-size-matters'
import { theme } from '~/constants'

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderRadius: 4
  },
  containerImage: {
    flex: 1,
    width: 96,
    flexDirection: 'column',
    backgroundColor: theme.darkGrey,
    borderRadius: 4,
    marginRight: 8
  },
  background: {
    flex: 1,
    backgroundColor: 'rgba(43, 43, 51, 0.6)',
    borderRadius: 4
  },
  containerCount: {
    flex: 1,
    marginRight: scale(9),
    borderRadius: 4
  },
  backgroundCount: {
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
    backgroundColor: 'rgba(43, 43, 51, 0.6)',
    borderRadius: 4
  },
  textCount: {
    fontStyle: 'normal',
    lineHeight: moderateScale(24),
    letterSpacing: 0,
    fontFamily: 'Mulish-Bold',
    fontSize: theme.fonts.MEDIUM_SIZE,
    color: theme.backgroundWhite
  },
  text: {
    fontStyle: 'normal',
    lineHeight: moderateScale(16),
    letterSpacing: 0,
    textAlign: 'left',
    fontFamily: 'Mulish-Bold',
    color: theme.backgroundWhite,
    marginHorizontal: scale(8),
    marginVertical: scale(8)
  },
  image: {
    width: 96,
    height: 96,
    borderRadius: 4
  }
})
