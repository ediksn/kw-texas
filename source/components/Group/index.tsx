import React from 'react'
import { View, Text, ImageBackground, TouchableOpacity } from 'react-native'
import { useTranslation } from 'react-i18next'
import { styles } from './styles'
import { GroupInterface } from '~/interfaces/groupInterface'
import { TEST_IDS } from '~/constants/testIds'

interface GroupProps {
  group: GroupInterface
  onGroupPress: (group: GroupInterface) => void
}

const Group = ({ group, onGroupPress }: GroupProps) => {
  const { icon, name } = group
  const { t } = useTranslation()

  return (
    <View style={styles.container} testID={TEST_IDS.GROUPS.GROUP_ID}>
      <TouchableOpacity style={styles.container} onPress={() => onGroupPress && onGroupPress(group)}>
        {icon?.url ? (
          <ImageBackground
            source={{ uri: icon.url }}
            imageStyle={styles.image}
            style={styles.containerImage}
            resizeMode='cover'
          >
            <View style={styles.background}>
              <Text style={styles.text} numberOfLines={4} ellipsizeMode='tail'>
                {name.trim().length > 1 ? name : t('components_General_NoName')}
              </Text>
            </View>
          </ImageBackground>
        ) : (
          <View style={styles.containerImage}>
            <View style={styles.background}>
              <Text style={styles.text} numberOfLines={4} ellipsizeMode='tail'>
                {name.trim().length > 1 ? name : t('components_General_NoName')}
              </Text>
            </View>
          </View>
        )}
      </TouchableOpacity>
    </View>
  )
}

export default Group
