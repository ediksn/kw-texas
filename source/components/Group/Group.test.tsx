import React from 'react'
import { render, RenderAPI } from '@testing-library/react-native'
import Group from '~/components/Group/index'
import { groupMockObject } from '../../../__mocks__/mockObjects'
import { TEST_IDS } from '~/constants/testIds'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

describe('Render Group Test', () => {
  let rendered: RenderAPI
  const mockOnPressGroup = jest.fn()

  beforeEach(() => {
    rendered = render(<Group group={groupMockObject} onGroupPress={mockOnPressGroup} />)
  })

  it('should render component Group correctly', () => {
    const component = rendered.getAllByTestId(TEST_IDS.GROUPS.GROUP_ID)
    expect(component).not.toBeNull()
  })

  it('Match with the snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('Renders correctly', () => {
    expect(rendered).toBeDefined()
  })
})
