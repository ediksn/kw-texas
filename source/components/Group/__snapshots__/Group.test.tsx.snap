// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Group Test Match with the snapshot 1`] = `
<View
  style={
    Object {
      "borderRadius": 4,
      "flex": 1,
    }
  }
  testID="groupId"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 4,
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "#282B33",
          "borderRadius": 4,
          "flex": 1,
          "flexDirection": "column",
          "marginRight": 8,
          "width": 96,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "https://avatar.kwconnect.com/556397.jpeg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": 96,
            },
            Object {
              "borderRadius": 4,
              "height": 96,
              "width": 96,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(43, 43, 51, 0.6)",
            "borderRadius": 4,
            "flex": 1,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={4}
          style={
            Object {
              "color": "white",
              "fontFamily": "Mulish-Bold",
              "fontStyle": "normal",
              "letterSpacing": 0,
              "lineHeight": 25.142857142857142,
              "marginHorizontal": 17.142857142857142,
              "marginVertical": 17.142857142857142,
              "textAlign": "left",
            }
          }
        >
          Test
        </Text>
      </View>
    </View>
  </View>
</View>
`;
